File: key-value-search/python/reverse.key.value.post.py
try:
    import http.client as http
except ImportError:
    import httplib as http

import json

username = 'Your reverse whois api username'
password = 'Your reverse whois api password'

payload = {
    'terms': [
        {
            'section': 'Admin',
            'attribute': 'name',
            'value': 'whois',
            'matchType': 'anywhere',
            'exclude': False
        }
    ],
    'recordsCounter': False,
    'username': username,
    'password': password,
    'outputFormat': 'json',
    'mode': 'purchase',
    'rows': 10
}


def print_response(txt):
    response_json = json.loads(txt)
    print(json.dumps(response_json, indent=4, sort_keys=True))


headers = {
    'Content-Type': 'application/json',
    'Accept': 'application/json'
}

conn = http.HTTPSConnection('www.whoisxmlapi.com')

conn.request('POST',
             '/reverse-whois-api/search.php',
             json.dumps(payload),
             headers)

response = conn.getresponse()
text = response.read().decode('utf8')

print_response(text)

File: key-value-search/php/reverse.key.value.post.php
<?php

$user = 'Your_reverse_whois_api_username';
$password = 'Your_reverse_whois_api_password';

$header = "Content-Type: application/json\r\nAccept: application/json\r\n";

$options = array('http' => array(
    'method' => 'POST',
    'header' => $header
));

$url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php';

$options['http']['content'] = json_encode(array(
    'terms'=>array(
        array(
            'section' => 'Registrant',
            'attribute' => 'Email',
            'value' => 'a@b.com',
            'exclude' => false
        )
    ),
    'recordsCounter' => false,
    'username' => $user,
    'password' => $password
));

print(file_get_contents($url, false, stream_context_create($options)));